lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -28,21 +39,162 @@ defmodule Mix.Tasks.Escript.Install do
 
   ## Command line options
 
-    * `--sha512` - checks the escript matches the given sha512 checksum
+    * `--sha512` - checks the escript matches the given sha512 checksum. Only
+      applies to installations via URL or local path.
 
     * `--force` - forces installation without a shell prompt; primarily
       intended for automation in build systems like make
 
+    * `--submodules` - fetch repository submodules before building escript from
+      git or github
+
   """
 
   @behaviour Mix.Local.Installer
 
   @escript_file_mode 0o555 # only read and execute permissions
 
-  @switches [force: :boolean, sha512: :string]
+  @switches [force: :boolean, sha512: :string, submodules: :boolean]
   @spec run(OptionParser.argv) :: boolean
   def run(argv) do
-    Mix.Local.Installer.install({__MODULE__, :escript}, argv, @switches)
+    {opts, args} = OptionParser.parse!(argv, strict: @switches)
+
+    case args do
+      ["git" | rest] ->
+        raise_if_sha512("git", opts)
+        install_from_git(rest, opts)
+      ["github" | rest] ->
+        raise_if_sha512("github", opts)
+        install_from_github(rest, opts)
+      ["hex" | rest] ->
+        raise_if_sha512("hex", opts)
+        install_from_hex(rest, opts)
```

Comment:
It shouldn't be an error. git, github and hex are valid paths to install escripts from.